src/script/api/script_text.cpp

```diff
@@ -248,8 +248,6 @@ void ScriptText::_GetEncodedText(std::back_insert_iterator<std::string> &output,
 
 const std::string Text::GetDecodedText()
 {
-	const std::string &encoded_text = this->GetEncodedText();
-
-	::SetDParamStr(0, encoded_text);
+	::SetDParamStr(0, this->GetEncodedText());
```

Comment:
For yet unknown reason, MSVC uses `void SetDParamStr(size_t n, const std::string &str)` for this call.

Reply:
`GetEncodedText` returned `const std::string`, which means MSVC does not see it as a temporary/rvalue. By making `GetEncodedText` just return `std::string`
Note that returning `const std::string` was common practice before C++11, however since it has become less common as it hinders move semantics. Also, the language has become more strict what must be done with respect to copy elision. This copy elision seems to have been one of the reasons to return const objects.